nixos/modules/config/users-groups.nix

```
@@ -592,6 +592,26 @@ in {
       '';
     };
 
+    # Warn about user accounts with deprecated password hashing schemes
+    system.activationScripts.hashes = {
+      deps = [ "users" ];
+      text = ''
+        users=()
+        while IFS=: read -r user hash tail; do
+          if [[ "$hash" != '!' && ! "$hash" =~ ^\$(y|gy|7|2b|2y|2a|6)\$ ]]; then
+            users+=("$user")
+          fi
+        done </etc/shadow
+
+        if [[ -n "''${users[@]}" ]]; then
```

It's technically valid to use empty usernames or usernames with spaces.
```suggestion
        if (( "''${#users[@]}" != 0 )); then
```

---

Apparently just `(( ${#users[@]} ))` is sufficient as a condition.